### PR TITLE
Small updates of CAP-58

### PR DESCRIPTION
**Pull request by dmkozh**
- Update for supporting v1 'create contract' auth payload 
- Clarify the semantics of CREATE_CONTRACT_HOST_FN w.r.t constructor support.

### CODE REVIEW

File: core/cap-0058.md [hunk 1]
```diff
@@ -277,11 +277,11 @@ Contract re-initialization is risky for a lot of cases, e.g. for contracts with
 
 While in some cases additional initialization may be required after Wasm has been updated, we deem the risks of allowing it to be higher than the benefits for such cases. Unlike initialization, atomicity is not as important, because the authorization priviliges don't change between the call to update the contract and the call to re-initialize it. Thus a programmatic, contract-specific solution should be feasible.
 
-### `SOROBAN_AUTHORIZED_FUNCTION_TYPE_CREATE_CONTRACT_HOST_FN` deprecation
+### `SOROBAN_AUTHORIZED_FUNCTION_TYPE_CREATE_CONTRACT_HOST_FN` is no longer supported
```
Comment (leighmcculloch) [inline core/cap-0058.md:280]: I started a new discussion thread on this topic:
- https://github.com/stellar/stellar-protocol/discussions/1501#discussioncomment-10423791

**Review by leighmcculloch — APPROVED** [review]

**Review by leighmcculloch — DISMISSED** [review]
I started a new discussion thread on the topic of the auth payload no support change:
- https://github.com/stellar/stellar-protocol/discussions/1501#discussioncomment-10423791

**Comment by leighmcculloch** [comment]
Does this need merging still?

**Comment by dmkozh** [comment]
> Does this need merging still?

Yes, but I've just realized that I've pushed the update to the wrong branch. This should PR be up-to-date now (i.e. specify that the old auth payload is still going to be supported).

**Review by leighmcculloch — APPROVED** [review]

**Pull request merged**